examples/keras_io/vision/image_classification_using_global_context_vision_transformer.py

```diff
@@ -0,0 +1,1182 @@
+            use_bias=False,
+            name="reduction",
+        )
+        self.norm1 = keras.layers.LayerNormalization(
+            -1, 1e-05, name="norm1"
+        )  # eps like PyTorch
+        self.norm2 = keras.layers.LayerNormalization(-1, 1e-05, name="norm2")
+
+    def call(self, inputs, **kwargs):
+        x = self.norm1(inputs)
+        xr = self.pad1(x)
+        for layer in self.conv:
+            xr = layer(xr)
+        x = x + xr
+        x = self.pad2(x)
+        x = self.reduction(x)
+        x = self.norm2(x)
+        return x
+
+
+class MLP(keras.layers.Layer):
+    """Multi-Layer Perceptron (MLP) block.
+
+    Args:
+        hidden_features: hidden features dimension.
+        out_features: output features dimension.
+        activation: activation function.
+        dropout: dropout rate.
+    """
+
+    def __init__(
+        self,
+        hidden_features=None,
+        out_features=None,
+        activation="gelu",
+        dropout=0.0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.hidden_features = hidden_features
+        self.out_features = out_features
+        self.activation = activation
+        self.dropout = dropout
+
+    def build(self, input_shape):
+        self.in_features = input_shape[-1]
+        self.hidden_features = self.hidden_features or self.in_features
+        self.out_features = self.out_features or self.in_features
+        self.fc1 = keras.layers.Dense(self.hidden_features, name="fc1")
+        self.act = keras.layers.Activation(self.activation, name="act")
+        self.fc2 = keras.layers.Dense(self.out_features, name="fc2")
+        self.drop1 = keras.layers.Dropout(self.dropout, name="drop1")
+        self.drop2 = keras.layers.Dropout(self.dropout, name="drop2")
+
+    def call(self, inputs, **kwargs):
+        x = self.fc1(inputs)
+        x = self.act(x)
+        x = self.drop1(x)
+        x = self.fc2(x)
+        x = self.drop2(x)
+        return x
+
+
+"""
+### Stem
+
+> **Notes**: In the code, this module is referred to as **PatchEmbed** but on paper, it
+is referred to as **Stem**.
+
+In the model, we have first used `patch_embed` module. Let's try to understand this
+module. As we can see from the `call` method,
+1. This module first **pads** input
+2. Then uses **convolutions** to extract patches with embeddings.
+3. Finally, uses `ReduceSize` module to first extract features with **convolution** but
+neither reduces spatial dimension nor increases spatial dimension.
+4. One important point to notice, unlike **ViT** or **SwinTransformer**, **GCViT**
+creates **overlapping patches**. We can notice that from the code,
+`Conv2D(self.embed_dim, kernel_size=3, strides=2, name='proj')`. If we wanted
+**non-overlapping** patches then we would've used the same `kernel_size` and `stride`.
+5. This module reduces the spatial dimension of input by $4x$.
+> **Summary:** image $\rightarrow$ padding $\rightarrow$ convolution $\rightarrow$
```

Comment:
Don't use LaTeX syntax, we don't render it. It you need an arrow just UTF-8, e.g. →